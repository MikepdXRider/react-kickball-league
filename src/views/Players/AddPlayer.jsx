import React from 'react'
import { useHistory } from 'react-router';
import { useEffect, useState } from 'react/cjs/react.development'
import PlayerForm from '../../components/PlayerForm/PlayerForm.jsx';
import { createPlayer } from '../../services/players.js';
import { getTeams } from '../../services/teams.js';

export default function AddPlayer() {
    const history = useHistory();

    const [playerNameStr, setPlayerNameStr] = useState('');
    const [playerPositionStr, setPlayerPositionStr] = useState('');
    const [playerTeamIdNum, setPlayerTeamIdNum] = useState(0);
    const [teamDataArr, setTeamDataArr] = useState([]);


    useEffect(() => {
        async function componentDidMount() {
            const newTeamDataArr = await getTeams();
            console.log(newTeamDataArr);
            setTeamDataArr(newTeamDataArr);
        }
        componentDidMount();
    }, [])


    const handleSubmit = async (e) => {
        e.preventDefault();
        // ðŸŸ¢ Get this working! When the user has not selected a valid team, do not make the createPlayer request.
        // if (playerTeamIdNum === '0'){
        //     prompt('Please assign player to a team.');
        //     return;
        // }
        const response = await createPlayer({name: playerNameStr, position: playerPositionStr,  teamId: playerTeamIdNum});
        history.push(`/players/${response[0].id}`);
    }

    return (
        <PlayerForm
            submitFn={handleSubmit}
            teamDataArr={teamDataArr}
            playerNameStr={playerNameStr}
            playerPositionStr={playerPositionStr}
            playerTeamIdNum={playerTeamIdNum}
            setPlayerNameStr={setPlayerNameStr}
            setPlayerPositionStr={setPlayerPositionStr}
            setPlayerTeamIdNum={setPlayerTeamIdNum}
        />
    )
}