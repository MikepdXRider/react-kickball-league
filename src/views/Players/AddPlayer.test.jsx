import { screen } from "@testing-library/react";
import { render } from "@testing-library/react";
import AddPlayer from "./AddPlayer.jsx";

it ('renders AddPlayer page', async () => {
    const { container } = render(<AddPlayer />)
    const text = await screen.findByText(/Add Player/i);

    expect(text).toBeInTheDocument();
    expect(container).toMatchSnapshot();
})