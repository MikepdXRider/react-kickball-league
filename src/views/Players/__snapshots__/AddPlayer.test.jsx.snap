// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AddPlayer page 1`] = `
<div>
  <form>
    <fieldset>
      <legend>
        Add Player
      </legend>
      <label
        for="name"
      >
        Name: 
      </label>
      <input
        id="name"
        name="Player-name"
        required=""
        type="text"
        value=""
      />
      <label
        for="position"
      >
        Position: 
      </label>
      <input
        id="position"
        name="Player-city"
        required=""
        type="text"
        value=""
      />
      <label
        for="team"
      >
        Team: 
      </label>
      <select
        id="team"
        name="teams"
        required=""
      >
        <option
          value=""
        >
          -
        </option>
      </select>
      <input
        type="submit"
      />
    </fieldset>
  </form>
</div>
`;
